src/1-raw-data/gcs/gcs.go

```diff
@@ -25,51 +25,60 @@ import (
 	"cloud.google.com/go/storage"
 	"context"
 	"fmt"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
 	"google.golang.org/api/iterator"
 	"io"
 	"log"
 	"net/http"
 )
 
+type GCSConnection interface {
+	StoreGCS(ctx context.Context, fileName, url string) error
+}
+
 type StorageConnection struct {
-	Ctx        context.Context
 	ProjectID  string
 	BucketName string
-	client     *storage.Client
-	bucket     *storage.BucketHandle
+	client     stiface.Client
+	bucket     stiface.BucketHandle
 }
 
-func (gcs *StorageConnection) ConnectGCSClient() error {
-	if client, err := storage.NewClient(gcs.Ctx); err != nil {
-		return fmt.Errorf("Failed to create client: %v", err)
+func (gcs *StorageConnection) ConnectGCSClient(ctx context.Context) error {
+	if c, err := storage.NewClient(ctx); err != nil {
+		return fmt.Errorf("Failed to create client: %w", err)
```

Comment:
If you want to test for this particular kind of error you could do something like
```
var (
  errCreateClient = errors.New("failed to create client")
)
```

then here:
```
return fmt.Errorf("%w: %v", errCreateClient, err)
```

Then your test becomes something like

```
got, gotErr := ConnectGCSClient(ctx)
if !errors.Is(gotErr, wantError) {
 ...
}
```

I don't think thats super useful here but its a good trick in general

Reply:
When I pull the strings out and away from the errors where they occur, it makes it not as easy to read the code. Looks like this is what you were going for with the %w you were telling me about. I am going to leave as is and then review when we meet next.